src/Elliptic/Systems/Poisson/Actions/Observe.hpp

```diff
@@ -0,0 +1,142 @@
+ *     \f$u^\mathrm{analytic}\f$
+ *   - `Poisson::Field::name() + "Error"`: The pointwise error
+ *     \f$u^\mathrm{numeric} - u^\mathrm{analytic}\f$
+ *   - `"InertialCoordinates_{x,y,z}"`
+ *
+ * Uses:
+ * - Metavariables:
+ *   - `analytic_solution_tag`
+ *   - Items required by `observers::Observer<Metavariables>`
+ * - DataBox:
+ *   - `LinearSolver::Tags::IterationId`
+ *   - `Tags::Mesh<Dim>`
+ *   - `Poisson::Field`
+ *   - `Tags::Coordinates<Dim, Frame::Inertial>`
+ *
+ * \note This action can be adjusted before compiling an executable to observe
+ * only the desired quantities.
+ */
+struct Observe {
+  template <typename... DbTags, typename... InboxTags, typename Metavariables,
+            size_t Dim, typename ActionList, typename ParallelComponent>
+  static auto apply(db::DataBox<tmpl::list<DbTags...>>& box,
+                    tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+                    Parallel::ConstGlobalCache<Metavariables>& cache,
+                    const ElementIndex<Dim>& array_index,
+                    const ActionList /*meta*/,
+                    const ParallelComponent* const /*meta*/) noexcept {
+    const auto& iteration_id = get<LinearSolver::Tags::IterationId>(box);
+    const auto& mesh = get<Tags::Mesh<Dim>>(box);
+    const std::string element_name = MakeString{} << ElementId<Dim>(array_index)
+                                                  << '/';
+
+    // Retrieve the current numeric solution
+    const auto& field = get<Poisson::Field>(box);
+
+    // Compute the analytic solution
+    const auto& inertial_coordinates =
+        db::get<::Tags::Coordinates<Dim, Frame::Inertial>>(box);
+    const auto field_analytic = get<Poisson::Field>(
+        Parallel::get<typename Metavariables::analytic_solution_tag>(cache)
+            .field_variables(inertial_coordinates));
+
+    // Compute error between numeric and analytic solutions
+    const DataVector field_error = get(field) - get(field_analytic);
+
+    // Compute l2 error squared over local element
+    const double local_l2_error_square = alg::accumulate(
+        field_error, 0.0, funcl::Plus<funcl::Identity, funcl::Square<>>{});
+
+    // Collect volume data
+    // Remove tensor types, only storing individual components
+    std::vector<TensorComponent> components;
+    components.reserve(3 + Dim);
+    components.emplace_back(element_name + Poisson::Field::name(), get(field));
+    components.emplace_back(element_name + Poisson::Field::name() + "Analytic",
+                            get(field_analytic));
+    components.emplace_back(element_name + Poisson::Field::name() + "Error",
+                            field_error);
+    components.emplace_back(element_name + "InertialCoordinates_x",
+                            get<0>(inertial_coordinates));
+    if (Dim >= 2) {
+      components.emplace_back(element_name + "InertialCoordinates_y",
+                              inertial_coordinates.get(1));
```

Comment:
oops, good point @gsb76 . This would only work if we had constexpr if's. My suggestion won't work without fancier tricks that won't be worth it for this code segment (`make_overloader` would do it, or if we could use the c++17 language features).